File: C++/NPC AI work/Finite State Machine/EnemyAI/States/EStateAttack.cpp
// Fill out your copyright notice in the Description page of Project Settings.


#include "EnemyAI/States/EStateAttack.h"

// Sets default values for this component's properties
UEStateAttack::UEStateAttack()
{

}

void UEStateAttack::OnEnter()
{
	UE_LOG(LogTemp, Warning, TEXT("Attack Entered"));
}

void UEStateAttack::SMTick()
{
	UE_LOG(LogTemp, Warning, TEXT("Attack Ticked"));
}

void UEStateAttack::OnExit()
{
	UE_LOG(LogTemp, Warning, TEXT("Attack Left"));
}

File: NPC AI work/Finite State Machine/C# Implementation/EStateAttack.cs
using System.Collections;
using UnityEditor;
using UnityEngine;

public class EStateAttack : IState
{
    Enemy_Basic parent;

    public EStateAttack(Enemy_Basic parent)
    {
        this.parent = parent;
    }

    public void OnEnter()
    {
        Debug.Log("Attack Entered");
    }

    public void OnExit()
    {
        Debug.Log("Attack Exited");
    }

    public void Tick()
    {
        Debug.Log("Attack Ticked");
    }
}